File: Segundo/PDOO/Tema 1/PeluqueriaRuby/lib/Citas.rb
require_relative "persona"
require_relative "peluqueria"

module Basico

    class Citas

        def initialize()
          @citas = Array.new
          @clientes = Array.new
        end

        def DarCita(persona)
            year= 2020+rand(3) #hasta 2 años a partir de 2020
            dia= 1+ rand(31) 
            hora= 9+rand(9)
            cita= Time.new(year, 11, dia, hora)
            @citas<<cita
            @clientes<<persona
            cita
        end

        def anularCita(persona)
            if (@clientes.include?(persona))
                indice = @clientes.find_index(persona)
                puts "Cita de "+persona.darNombre+" a las "+@citas[indice].to_s+" anulada"
                @citas.delete(indice)
                @clientes.delete(indice)
                if (@citas.empty?)
                    puts "No quedan más citas"
                end
            end
        end
    end
end

File: Segundo/PDOO/Tema 1/Peluqueria/src/basico/Citas.java

package basico;

import java.time.LocalDateTime;
import java.time.Month;
import java.util.ArrayList;


public class Citas {
    private ArrayList<LocalDateTime> citas = new ArrayList<>();
    private ArrayList<Persona> persona = new ArrayList<>();
    
    LocalDateTime DarCita(Persona p, Peluqueria pelu) {
        int year= (int)(Math.random()*2+2020);	//hasta 2 años a partir del 2020
        int dia= (int) (Math.random()*30+1);
        int hora= (int) (Math.random()*8+9);      
        LocalDateTime cita= LocalDateTime.of(year, Month.OCTOBER, dia, hora, 0);
        citas.add(cita);
        persona.add(p);
        return cita;
    }
    
    void anularCita(Persona p, Peluqueria pelu) {
        int indice = persona.indexOf(p);
        System.out.println("Cita de "+p.ConsultaNombre()+" a las "+citas.get(indice)+" anulada");
        citas.remove(indice);
        persona.remove(indice);
        if (citas.isEmpty()) {
            System.out.println("No hay mas citas en cola");
        }
    }
}
